devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java

```diff
@@ -265,7 +266,8 @@ public void execute() throws MojoFailureException, MojoExecutionException {
             return;
         }
 
-        if (!sourceDir.isDirectory()) {
+        // don't warn if running in a module of a multi-module project
+        if (!sourceDir.isDirectory() && !Optional.ofNullable(project.getParentFile()).map(File::exists).orElse(false)) {
```

Comment:
Could you please change the second part to `project.getParentFile() != null && project.getParentFile().exists()`? It's an lot more simpler and efficient expression.

Reply:
I prefer the more DRY `Optional` approach but sure, I can (and will) change it.

Reply:
I went with a slightly different approach with just one negation instead of two.